e2e-spock-geb/Jenkinsfile.template

```diff
@@ -0,0 +1,57 @@
+/* generated jenkins file used for building and deploying @component_id@ in projects @project_id@ */
+def final projectId = '@project_id@'
+def final componentId = '@component_id@'
+def final credentialsId = "${projectId}-cd-cd-user-with-password"
+def sharedLibraryRepository
+def dockerRegistry
+node {
+  sharedLibraryRepository = env.SHARED_LIBRARY_REPOSITORY
+  dockerRegistry = env.DOCKER_REGISTRY
+}
+
+library identifier: 'ods-library@@ods_git_ref@', retriever: modernSCM(
+  [$class: 'GitSCMSource',
+   remote: sharedLibraryRepository,
+   credentialsId: credentialsId])
+
+/*
+  See readme of shared library for usage and customization
+  @ https://github.com/opendevstack/ods-jenkins-shared-library/blob/master/README.md
+  eg. to create and set your own builder slave instead of 
+  the maven/gradle slave used here - the code of the slave can be found at
+  https://github.com/opendevstack/ods-quickstarters/tree/master/common/jenkins-slaves/maven
+ */ 
+odsPipeline(
+  image: "${dockerRegistry}/cd/jenkins-slave-maven:@ods_image_tag@",
+  projectId: projectId,
+  componentId: componentId,
+  branchToEnvironmentMapping: [
+    'master': 'test',
+    '*': 'dev'
+  ]
+) { context ->
+  stageScanForSonarqube(context)
+  stageTest(context)
+}
+
+def stageTest(def context) {
+  def javaOpts = "-Xmx512m"
+  def gradleTestOpts = "-Xmx128m"
+  def springBootEnv = context.environment
+  if (springBootEnv.contains('-dev')) {
+    springBootEnv = 'dev'
+  }
+
+  stage('Test') {
+    withEnv(["TAGVERSION=${context.tagversion}", "NEXUS_HOST=${context.nexusHost}", "NEXUS_USERNAME=${context.nexusUsername}", "NEXUS_PASSWORD=${context.nexusPassword}", "JAVA_OPTS=${javaOpts}","GRADLE_TEST_OPTS=${gradleTestOpts}","ENVIRONMENT=${springBootEnv}"]) {
+      def status = sh(script: "./gradlew clean test --stacktrace --no-daemon -Dtest.ignoreFailures=true", returnStatus: true)
```

Comment:
Let's remove the use of ignoreFailures from these lines and provide a central variable in build.gradle.

Reply:
Done